packages/memory/src/index.ts

```diff
@@ -152,13 +152,18 @@ export class Memory extends MastraMemory {
         if (selectBy?.last && v1Messages.length > selectBy.last) {
           // ex: 23 (v1 messages) minus 20 (selectBy.last messages)
           // means we will start from index 3 and keep all the later newer messages from index 3 til the end of the array
-          return v1Messages.slice(v1Messages.length - selectBy.last);
+          return v1Messages.slice(v1Messages.length - selectBy.last) as CoreMessage[];
         }
-        return v1Messages;
+        // TODO: this is absolutely wrong but became apparent that this is what we were doing before adding MessageList. Our public types said CoreMessage but we were returning MessageType which is equivalent to MastraMessageV1
+        // In a breaking change we should make this the type it actually is.
+        return v1Messages as CoreMessage[];
```

Comment:
I realized this is how the public types were before. We were returning mastra db records but casting them as CoreMessage